glusterd2/events/liveness.go

```diff
@@ -65,15 +65,28 @@ func (l *livenessWatcher) Watch() {
 	}
 }
 
+//Stop will stop the livenessWatcher if it is running and waits for
+//it to exit.
+func (l *livenessWatcher) Stop() error {
+	if l.stopCh == nil {
+		return errors.New("livness watcher has not been started")
+	}
+	close(l.stopCh)
+	l.stopCh = nil
```

Comment:
question: why we need to assign ```nil``` to channel after closing it?

Reply:
To guard against `Stop()` being called again. Closing a closed channel causes panic

Reply:
if someone tries to close stopCh again then it will not result in panic, because a nil check is already given at the start of the method body.

Reply:
we can do 
```
read,ok := <- l.stopCh
if !ok{
return errors.New("livness watcher has not been started")
}
```

Reply:
The goal of stop channel isn't to pass values around. The logically valid operation at the sender end is to close it.